lightblue-ldap-config/src/main/java/com/redhat/lightblue/config/ldap/LdapDBResolver.java

```diff
@@ -71,20 +70,20 @@ private LdapDataSourceConfiguration findByDatabase(String database){
         return null;
     }
 
-	@Override
-	public List<LDAPConnection> getConnections() {
-		
-		List<LDAPConnection> ldapConnections = new ArrayList<>();
-		
-		for (LdapDataSourceConfiguration ldapDS : ldapDataSources){
-			try {
-				ldapConnections.add(ldapDS.getLdapConnection());
-			} catch (LDAPException e) {
-				ldapConnections.add(null);
-			}
-		}
-		
-		return Collections.unmodifiableList(ldapConnections);
-	}
+    @Override
+    public Map<String, Object> getLDAPConnectionsStatus() {
+        
+        Map<String, Object> connectionsStatus = new HashMap<>();
+        LDAPConnection connection = null;
+        for (LdapDataSourceConfiguration ldapDS : ldapDataSources) {
+            try {
+                connection = ldapDS.getLdapConnection();
+                connectionsStatus.put(ldapDS.getDatabaseName(), connection.isConnected());
```

Comment:
ldapDataSources is a Set of LdapDataSourceConfiguration which identifies uniqueness on the basis of database name (based on equals and hashcode implementation). Hence, I believe it is safe to use database name as key to the Map.